nixos/modules/services/networking/corerad.nix

```diff
@@ -4,18 +4,58 @@ with lib;
 
 let
   cfg = config.services.corerad;
+
+  writeTOML = name: x:
+    pkgs.runCommandNoCCLocal name {} ''
+      echo '${builtins.toJSON x}' | ${pkgs.go-toml}/bin/jsontoml > $out
+    '';
+
+  # Prefer settings attributes over a config file if both are set.
+  coreradTOML = if cfg.settings != null then
+    writeTOML "corerad.toml" cfg.settings
+  else
+    cfg.configFile;
+
 in {
-  meta = {
-    maintainers = with maintainers; [ mdlayher ];
-  };
+  meta.maintainers = with maintainers; [ mdlayher ];
 
   options.services.corerad = {
     enable = mkEnableOption "CoreRAD IPv6 NDP RA daemon";
 
+    settings = mkOption {
+      default = null;
```

Comment:
Then you can remove the default here too, which means it will ask people to set a `settings` value.